dev_tools/nyaml2nxdl/nyaml2nxdl_backward_tools.py

```diff
@@ -0,0 +1,1046 @@
+
+        allowed_attr = [
+            "name",
+            "type",
+            "units",
+            "nameType",
+            "recommended",
+            "optional",
+            "minOccurs",
+            "maxOccurs",
+            "deprecated",
+        ]
+
+        name = ""
+        node_attr = node.attrib
+        if "name" in node_attr:
+            pass
+        else:
+            raise ValueError("Attribute must have an name key.")
+        rm_key_list = []
+        # Maintain order: name and type in form name(type) or (type)name that come first
+        for key, val in node_attr.items():
+            if key == "name":
+                name = val
+                rm_key_list.append(key)
+
+        for key in rm_key_list:
+            del node_attr[key]
+
+        file_out.write(
+            "{indent}{escapesymbol}{name}:\n".format(
+                indent=depth * DEPTH_SIZE, escapesymbol=r"\@", name=name
+            )
+        )
+
+        tmp_dict = {}
+        exists_dict = {}
+        for key, val in node_attr.items():
+            # As both 'minOccurs', 'maxOccurs' and optionality move to the 'exists'
+            if key in ["minOccurs", "maxOccurs", "optional", "recommended", "required"]:
+                if "exists" not in tmp_dict:
+                    tmp_dict["exists"] = []
+                self.handle_exists(exists_dict, key, val)
+            elif key == "units":
+                tmp_dict["unit"] = val
+            else:
+                tmp_dict[key] = val
+            if key not in allowed_attr:
+                raise ValueError(
+                    f"An attribute ({key}) has been found that is not allowed."
+                    f"The allowed attr is {allowed_attr}."
+                )
+
+        has_min_max = False
+        has_opt_reco_requ = False
+        if exists_dict:
+            for key, val in exists_dict.items():
+                if key in ["minOccurs", "maxOccurs"]:
+                    tmp_dict["exists"] = tmp_dict["exists"] + val
+                    has_min_max = True
+                elif key in ["optional", "recommended", "required"]:
+                    tmp_dict["exists"] = key
+                    has_opt_reco_requ = True
+        if has_min_max and has_opt_reco_requ:
+            raise ValueError(
+                "Optionality 'exists' can take only either from ['minOccurs',"
+                " 'maxOccurs'] or from ['optional', 'recommended', 'required']"
+                ". But not from both of the groups together. Please check in"
+                " attributes"
+            )
+
+        depth_ = depth + 1
+        for key, val in tmp_dict.items():
+            # Increase depth size inside handle_map...() for writting text with one
+            # more indentation.
+            file_out.write(
+                f"{depth_ * DEPTH_SIZE}{key}: "
+                f"{handle_mapping_char(val, depth_ + 1, False)}\n"
+            )
+
+    def handel_link(self, depth, node, file_out):
```

Comment:
Use a non-musical name: "handle_link", here and other methods below

Reply:
there was a spelling mistake in `handle_...`. fixed!